### PR TITLE
fix 2.0: use default transport to mock http requests

### PR DESCRIPTION
**Pull request by MatousJobanek**
gock relies on the default transport that mocks

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -55,6 +55,14 @@ func AddToScheme() error {
...
 
 func NewClient(token, apiEndpoint string) (runtimeclient.Client, error) {
+	return NewClientWitTransport(token, apiEndpoint, &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true, // nolint: gosec
+		},
+	})
+}
+
+func NewClientWitTransport(token, apiEndpoint string, transport http.RoundTripper) (runtimeclient.Client, error) {
```
Comment (xcoulon) [inline pkg/client/client.go:65]:
```suggestion
func NewClientWithTransport(token, apiEndpoint string, transport http.RoundTripper) (runtimeclient.Client, error) {
```

Comment (xcoulon) [reply pkg/client/client.go:65]: (small typo)

Comment (MatousJobanek) [reply pkg/client/client.go]: Friday's morning a2a3c2f

Comment (MatousJobanek) [reply pkg/client/client.go]: And thanks for the catch ;-)

Comment (xcoulon) [reply pkg/client/client.go]: my pleasure! 😇

Comment (MatousJobanek) [reply pkg/client/client.go]: can you click on approve now? or is there anything still missing?

**Review by xcoulon — APPROVED** [review]
sorry for the late approval :/

**Pull request merged**